nx/lib/nx.ex

```diff
@@ -1713,6 +1726,15 @@ defmodule Nx do
   def to_binary(tensor, opts \\ []) do
     opts = keyword!(opts, [:limit])
     tensor = to_tensor(tensor)
+
+    # TODO: add proper implementation
```

Comment:
What would be the proper implementation? In any case, I think we can add `flat_size`, which is the size including any vectorized dimension. Maybe that's what you mean?

Reply:
Yes, exactly this. As it stands, `size` will return the base shape's size, so we need to devectorize the tensor.
`flat_size` seems like a nice addition